File: packages/honker-rs/examples/basic.rs
//! Basic example: enqueue a few jobs, claim them, ack each.
//!
//! cargo run --example basic

use honker::{Database, EnqueueOpts, QueueOpts};
use serde_json::json;

fn main() -> honker::Result<()> {
    let db = Database::open("demo.db")?;
    let q = db.queue("emails", QueueOpts::default());

    for i in 0..3 {
        q.enqueue(
            &json!({"to": format!("user-{}@example.com", i)}),
            EnqueueOpts::default(),
        )?;
    }

    while let Some(job) = q.claim_one("worker-1")? {
        let body: serde_json::Value = job.payload_as()?;
        println!("processing job {}: to={}", job.id, body["to"]);
        job.ack()?;
    }

    Ok(())
}

File: packages/honker-bun/examples/basic.ts
// bun run examples/basic.ts
import { open } from "../src/index.ts";

const EXT_PATH =
  process.env.HONKER_EXTENSION_PATH ??
  new URL(
    "../../../target/release/libhonker_extension.dylib",
    import.meta.url,
  ).pathname;

const db = open("demo.db", EXT_PATH);
const q = db.queue("emails");

for (let i = 0; i < 3; i++) {
  q.enqueue({ to: `user-${i}@example.com` });
}

while (true) {
  const job = q.claimOne("worker-1");
  if (!job) break;
  console.log(`processing job ${job.id}: to=${(job.payload as any).to}`);
  job.ack();
}

db.close();
